### PR TITLE
VirtualizedList: fix bug where onViewableItemsChanged wouldn't trigger

### PR DESCRIPTION
**Pull request by formatlos**
In the current implementation of the `VirtualizedList` the `onViewableItemsChanged` callback wouldn't trigger if the underlying list data changes. (see example snack https://snack.expo.io/Hk5703eBb)

I added a method in the `ViewabilityHelper` to invalidate the cached viewableIndices, which gets triggered when the list-data changes.

### CODE REVIEW

**Comment by formatlos** [comment]
Do you have any thoughts on this, I really need some way to make the `VirtualizedList` work as expected.

File: Libraries/Lists/__tests__/VirtualizedList-test.js [hunk 1]
```diff
@@ -159,4 +159,57 @@ describe('VirtualizedList', () => {
...
+    expect(onViewableItemsChanged).toHaveBeenCalledTimes(1);
+    expect(onViewableItemsChanged).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        viewableItems: [expect.objectContaining({isViewable: true, key: 'i1'})],
+      }),
+    );
+    data = [{key: 'i4'}, ...data];
+    component.update(<VirtualizedList {...props} data={data} />);
+
+    instance._onScroll(scrollEvent);
```
Comment (sahrens) [inline Libraries/Lists/__tests__/VirtualizedList-test.js]: Hmm, perhaps you should change the scroll event offset so it's more realistic? I could imagine adding some optimization that filters out scroll events that are identical and this test breaking.

**Comment by formatlos** [comment]
Still any concerns about this PR? If not I'd really appreciate if you could merge it in.

**Comment by sahrens** [comment]
Looks like CI checks are failing - can you rebase, fix flow and make sure everything is passing and I'll try to import it again?

**Comment by formatlos** [comment]
all tests are running locally. The failing test doesn't seem to be related to my changes.

**Comment by formatlos** [comment]
Did the import succeed?

**Pull request closed without merging**